### PR TITLE
feat(cheatcodes): fork cheatcode

### PR DESCRIPTION
**Pull request by Deniallugo**
# What :computer: 
* Adds support for fork cheatcodes

# Why :hand:
* More support for cheatcodes

# Notes :memo:
* Requires https://github.com/matter-labs/era-test-node/pull/251 to be merged before this PR and the dep `era-test-node` to be updated to correct rev

### CODE REVIEW

File: crates/era-cheatcodes/tests/src/cheatcodes/Addr.t.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        uint256 pk = 77814517325470205911140941194401928579557062014761831930645393041380819009408;
+        address expected = 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266;
+
+        (bool success, bytes memory data) = Constants.CHEATCODE_ADDRESS.call(
+            abi.encodeWithSignature("addr(uint256)", pk)
+        );
+        require(success, "addr failed");
+        address addr = abi.decode(data, (address));
+        assertEq(addr, expected, "expected address did not match");
+        console.log("failed?", failed());
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply crates/era-cheatcodes/tests/src/cheatcodes/Addr.t.sol:18]: hm, weird, that's it's even in review... probably wrong copypaste

File: crates/evm/core/src/era_revm/db.rs [hunk 2]
```diff
@@ -25,27 +25,30 @@ use zksync_basic_types::{
...
-    SYSTEM_CONTEXT_ADDRESS,
+    SYSTEM_CONTEXT_ADDRESS, SYSTEM_CONTEXT_CURRENT_L2_BLOCK_INFO_POSITION,
 };
 
 use zksync_utils::{address_to_h256, h256_to_u256, u256_to_h256};
 
 #[derive(Default)]
 pub struct RevmDatabaseForEra<DB> {
     pub db: Arc<Mutex<Box<DB>>>,
+    // pub env: Arc<Mutex<Env>>,
```
Comment (Deniallugo) [inline crates/evm/core/src/era_revm/db.rs:36]: commented code. 

Is it enough now to have only the current bloc?

**Review by Deniallugo — COMMENTED** [review]
Cool! I like, that we removed a lot of cludges eventually

File: crates/evm/core/src/era_revm/db.rs (hunk 2 above)
Comment (nbaztec) [reply crates/evm/core/src/era_revm/db.rs:36]: Yeah, the `env` isn't tracked properly by foundry anyway, e.g. for the block after setUp the changes aren't applied. I figured previously it was easier to simply rely on storage.

**Review by nbaztec — APPROVED** [review]

**Pull request merged**